vizro-core/tests/unit/vizro/models/_action/test_action.py

```diff
@@ -48,47 +102,30 @@ def test_create_action_mandatory_only(self, identity_action_function):
         assert action.inputs == []
         assert action.outputs == []
 
-    def test_create_action_mandatory_and_optional(self, identity_action_function):
-        function = identity_action_function()
-        inputs = ["component_1.property_A", "component_1.property_B"]
-        outputs = ["component_2.property_A", "component_2.property_B"]
-
-        action = Action(function=function, inputs=inputs, outputs=outputs)
-
-        assert hasattr(action, "id")
-        assert action.function is function
-        assert action.inputs == inputs
-        assert action.outputs == outputs
-
-    def test_is_model_inheritable(self, identity_action_function):
-        class MyAction(vm.Action):
-            pass
-
-        function = identity_action_function()
-        my_action = MyAction(function=function)
-
-        assert hasattr(my_action, "id")
-        assert my_action.function is function
-        assert my_action.inputs == []
-        assert my_action.outputs == []
+        # TODO: Should we test private attributes?
+        assert action._legacy is False
+        assert action._transformed_inputs == {}
+        assert action._transformed_outputs == []
+        assert action._dash_components == []
+        assert action._action_name == "_identity_action_function"
 
     @pytest.mark.parametrize(
-        "inputs, outputs",
+        "inputs, _transformed_inputs",
         [
             ([], []),
-            (["component.property"], ["component.property"]),
-            (["component.property", "component.property"], ["component.property", "component.property"]),
+            (["component.property"], [State("component", "property")]),
             (
-                ["Component_1.Property_A", "Component_2.Property_B"],
-                ["Component_1.Property_A", "Component_2.Property_B"],
+                ["component.property", "component.property"],
+                [State("component", "property"), State("component", "property")],
             ),
         ],
     )
-    def test_inputs_outputs_valid(self, inputs, outputs, identity_action_function):
-        action = Action(function=identity_action_function(), inputs=inputs, outputs=outputs)
+    def test_model_field_inputs_valid(self, identity_action_function, inputs, _transformed_inputs):
+        action = Action(function=identity_action_function(), inputs=inputs)
 
+        assert action._legacy is True
```

Comment:
```suggestion
        assert action._legacy
```

Reply:
And `assert not action._legacy` instead of `is False` also. I know this is a weaker check but it just looks weird to me currently and we know that `legacy` is just a bool.